android/src/main/java/com/reactnativespokestack/SpokestackModule.kt

```diff
@@ -0,0 +1,287 @@
+import io.spokestack.spokestack.Spokestack
+import io.spokestack.spokestack.profile.*
+import io.spokestack.spokestack.tts.AudioResponse
+import io.spokestack.spokestack.tts.SpokestackTTSOutput
+import io.spokestack.spokestack.tts.SynthesisRequest
+import javax.annotation.Nullable
+
+class SpokestackModule(private val reactContext: ReactApplicationContext): ReactContextBaseJavaModule(reactContext) {
+  private val adapter = SpokestackAdapter { event: String, data: WritableMap -> sendEvent(event, data) }
+  private lateinit var spokestack: Spokestack
+  private val promises = mutableMapOf<String, Promise>()
+  private var say: ((url: String) -> Unit)? = null
+  private lateinit var audioPlayer: SpokestackTTSOutput
+
+  override fun getName(): String {
+    return "Spokestack"
+  }
+
+  override fun onCatalystInstanceDestroy() {
+    super.onCatalystInstanceDestroy()
+    if (spokestack.speechPipeline != null) {
+      spokestack.stop()
+    }
+  }
+
+  private enum class PipelineProfiles(p:Class<out PipelineProfile>) {
+    TFLiteWakewordNativeASR(TFWakewordAndroidASR::class.java),
+    VADNativeASR(VADTriggerAndroidASR::class.java),
+    PTTNativeASR(PushToTalkAndroidASR::class.java),
+    TFLiteWakewordSpokestackASR(TFWakewordSpokestackASR::class.java),
+    VADSpokestackASR(VADTriggerSpokestackASR::class.java),
+    PTTSpokestackASR(PushToTalkSpokestackASR::class.java);
+    private val profile:Class<out PipelineProfile> = p
+    fun value():String {
+      return this.profile.name
+    }
+  }
+
+  private fun sendEvent(event:String, @Nullable params:WritableMap) {
+    if (reactContext.hasActiveCatalystInstance()) {
+      when(event.toLowerCase()) {
+        "error" -> {
+          // Reject all existing promises
+          for ((key, promise) in promises) {
+            promise.reject(Exception("Error in Spokestack during $key."))
+          }
+          promises.clear()
+        }
+        "synthesize" -> {
+          val url = params.getString("url")
+          promises["synthesize"]?.resolve(url)
+          promises.remove("synthesize")
+          if (say != null && url != null) {
+            say!!(url)
+            say = null
+          }
+        }
+        "classify" -> {
+          promises["classify"]?.resolve(params.getMap("result"))
+          promises.remove("classify")
+        }
+        else -> println("Sending JS event: $event")
+      }
+      reactContext
+        .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+        .emit(event.toLowerCase(), params)
+    }
+  }
+
+  private fun kebabCase(str: String): String {
+    return str.replace(Regex("([a-z])([A-Z])")) {
+      m -> "${m.groupValues[1]}-${m.groupValues[2].toLowerCase()}"
+    }
+  }
+
+  private fun textToSpeech(input: String, format: Int, voice: String) {
+    if (format > 2 || format < 0) {
+      throw Exception(("A format of $format is not supported. Please use an int between 0 and 2. Refer to documentation for further details."))
```

Comment:
> Pedantic, but would "from 0 to 2" be better? This could be read as "1 is the only valid value".

Sure, but this was just copy/paste from what was there before.

> Could there be an enum in TS that uses ordinal values under the hood as a bridge?

This is already the case. See TTSFormat.

Reply:
Heh, yeah; realized the second one when I read through the JS piece. I guess my comment reduces to "could we change the error message?".